src/Microsoft.AspNetCore.Rewrite/Internal/CodeRules/RedirectRule.cs

```diff
@@ -38,9 +48,17 @@ public override void ApplyRule(RewriteContext context)
             {
                 var newPath = initMatchResults.Result(Replacement);
                 var response = context.HttpContext.Response;
+
                 response.StatusCode = StatusCode;
+                context.Result = RuleTermination.ResponseComplete;
+
+                if (string.IsNullOrEmpty(newPath))
```

Comment:
It should never be null, but I think the edge case is if a user captures an empty string (which I'm not sure is possible), they can have an empty string.